File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int,int> umap;
        for (int i = 0; i<n; i++){
            int diff = target-nums[i];
            if (umap.find(diff)==umap.end())
                umap[nums[i]] = i;
            else return {i, umap[diff]};
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        umap = {}
        for i in range(n):
            diff = target - nums[i]
            if diff not in umap:
                umap[nums[i]] = i
            else: return [i, umap[diff]]
        return []
        